packages/bun-usockets/src/loop.c

```diff
@@ -422,27 +424,36 @@ void us_internal_dispatch_ready_poll(struct us_poll_t *p, int error, int events)
                         #undef LOOP_ISNT_VERY_BUSY_THRESHOLD
                         #endif
                     } else if (!length) {
-                        if (us_socket_is_shut_down(0, s)) {
-                            /* We got FIN back after sending it */
-                            /* Todo: We should give "CLEAN SHUTDOWN" as reason here */
-                            s = us_socket_close(0, s, LIBUS_SOCKET_CLOSE_CODE_CLEAN_SHUTDOWN, NULL);
-                            return;
-                        } else {
-                            /* We got FIN, so stop polling for readable */
-                            us_poll_change(&s->p, us_socket_context(0, s)->loop, us_poll_events(&s->p) & LIBUS_SOCKET_WRITABLE);
-                            s = s->context->on_end(s);
-                        }
+                        eof = 1; // lets handle EOF in the same place
```

Comment:
Is this always write EOF and never write EOF?

Reply:
this is the reading loop so is always receiving FIN/EOF